src/jllvm/vm/StringInterner.cpp

```diff
@@ -0,0 +1,66 @@
+#include "StringInterner.hpp"
+
+const jllvm::ClassObject* jllvm::StringInterner::getStringClassObject()
+{
+    if (!m_stringClass)
+    {
+        m_stringClass = &m_classLoader.forName(stringDescriptor);
+        checkStructure();
+    }
+    return m_stringClass;
+}
+
+void jllvm::StringInterner::checkStructure()
+{
+    for (const auto& item : m_stringClass->getFields())
+    {
+        if (!item.isStatic())
```

Comment:
You can reduce indentation here via an early continue:
```cpp
if (item.isStatic()) 
{
    continue;
}

...
```
I also think clang-tidy apparently doesn't see the asserts, hence thinks these are identical branches.
Maybe use `#ifdef NDEBUG` around the whole implementation so that the function body is empty if we are in a release build?